dali/test/python/test_operator_transpose.py

```diff
@@ -78,19 +81,25 @@ def iter_setup(self):
         data = self.iterator.next()
         self.feed_input(self.data, data, layout=self.layout)
 
-def check_transpose_vs_numpy(device, batch_size, shape, permutation):
-    eii1 = RandomDataIterator(batch_size, shape=shape)
-    eii2 = RandomDataIterator(batch_size, shape=shape)
-    compare_pipelines(TransposePipeline(device, batch_size, "HWC", iter(eii1), permutation=permutation),
-                      PythonOpPipeline(lambda x: transpose_func(x, permutation), batch_size, "HWC", iter(eii2)),
-                      batch_size=batch_size, N_iterations=10)
+def check_transpose_vs_numpy(device, batch_size, dim, total_volume, permutation):
+    max_shape = [int(math.pow(total_volume/batch_size, 1/dim))] * dim
+    print("Testing", device, "backend with batch of", batch_size, "max size", max_shape)
+    print("permutaiton ", permutation)
+    eii1 = RandomlyShapedDataIterator(batch_size, max_shape=max_shape)
+    eii2 = RandomlyShapedDataIterator(batch_size, max_shape=max_shape)
+    compare_pipelines(TransposePipeline(device, batch_size, "", iter(eii1), permutation=permutation),
+                      PythonOpPipeline(lambda x: transpose_func(x, permutation), batch_size, "", iter(eii2)),
+                      batch_size=batch_size, N_iterations=5)
+
+def all_permutations(n):
+    return itertools.permutations(range(n))
 
 def test_transpose_vs_numpy():
-    for device in {'cpu', 'gpu'}:
-        for batch_size in {1, 3}:
-            for shape in {(2048, 512, 1), (2048, 512, 3), (2048, 512, 8)}:
-                for permutation in ([0, 1, 2], [0, 2, 1], [1, 0, 2], [1, 2, 0], [2, 0, 1], [2, 1, 0]):
-                    yield check_transpose_vs_numpy, device, batch_size, shape, permutation
+    for device in ['cpu', 'gpu']:
```

Comment:
How fast/slow test are after that change?

Reply:
The old tests took 100s with new transpose and 103s with old transpose.
The new tests took 58s with old transpose and 24s with new transpose - and that includes all the time needed to run reference pipeline (in numpy!) and all cpu tests - so the difference is immense.

Reply:
Understood.

Reply:
This difference is when having to rebuild the cutt plan or even without having to do so?

Reply:
Yes, the old test didn't rebuild anything (and was actually running one transpose, with the whole batch treated as a tensor with extra dim).